testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,469 @@
+        yield f
+
+
+@pytest.mark.parametrize('fname, xdr', ((XTC_multi_frame, XTCFile),
+                                        (TRR_multi_frame, TRRFile)))
+class TestCommonAPI:
+    @staticmethod
+    @pytest.fixture
+    def reader(fname, xdr):
+        with xdr(fname) as f:
+            yield f
+
+    def test_natoms(self, reader):
+        assert reader.n_atoms == 10
+
+    def test_over_seek(self, reader):
+        with pytest.raises(EOFError):
+            reader.seek(100)
+
+    def test_read_after_iteration(self, reader):
+        for frame in reader:
+            pass
+        with pytest.raises(EOFError):
+            reader.read()
+
+    def test_zero_based_frame_tell(self, reader):
+        assert reader.tell() == 0
+
+    def test_read_closed(self, reader):
+        reader.close()
+        with pytest.raises(IOError):
+            reader.read()
+
+    def test_iter(self, reader):
+        for i, frame in enumerate(reader):
+            assert reader.tell() == i + 1
+
+    def test_double_iter(self, reader):
+        for i, frame in enumerate(reader):
+            assert reader.tell() == i + 1
+        for i, frame in enumerate(reader):
+            assert reader.tell() == i + 1
+
+    def test_seek(self, reader):
+        reader.seek(4)
+        assert reader.tell() == 4
+
+    def test_seek_negatice(self, reader):
+        with pytest.raises(IOError):
+            reader.seek(-4)
+
+    def test_len(self, reader):
+        assert len(reader) == 10
+        # ensure we didn't go to another frame
+        assert reader.tell() == 0
+
+    def test_raise_not_existing(self, xdr, fname):
+        with pytest.raises(IOError):
+            xdr('foo')
+
+    def test_open_wrong_mode(self, xdr, fname):
+        with pytest.raises(IOError):
+            xdr('foo', 'e')
+
+    def test_read_write_mode_file(self, xdr, tmpdir, fname):
+        fname = str(tmpdir.join('foo'))
+        with xdr(fname, 'w') as f:
+            with pytest.raises(IOError):
+                f.read()
+
+
+@pytest.mark.parametrize("xdrfile, fname, offsets",
+                         ((XTCFile, XTC_multi_frame, XTC_OFFSETS),
+                          (TRRFile, TRR_multi_frame, TRR_OFFSETS)))
+class TestOffsets:
+    @staticmethod
+    @pytest.fixture
+    def reader(xdrfile, fname):
+        with xdrfile(fname) as f:
+            yield f
+
+    def test_offset(self, reader, offsets):
+        assert_array_equal(reader.offsets, offsets)
+
+    def test_set_offsets(self, reader, offsets):
+        reader.set_offsets(np.arange(len(offsets)))
+        assert_array_equal(reader.offsets, np.arange(len(offsets)))
+
+    def test_fail_wrong_offsets(self, reader, offsets):
+        reader.set_offsets(np.arange(len(offsets)))
+        reader.seek(5)
+        with pytest.raises(IOError):
+            reader.read()
+
+    def test_seek_overflow(self, reader, offsets):
+        with pytest.raises(OverflowError):
+            reader._bytes_seek(2**65)
```

Comment:
there is  a bad side effect though. We need to always use all parametrized values to a function even if they are not used.